src/Appwrite/Utopia/Request/Filters/V12.php

```diff
@@ -0,0 +1,156 @@
+                $content = $this->addCollectionPermissionLevel($content);
+                break;
+            case "database.updateCollection":
+                $content = $this->removeRules($content);
+                $content = $this->addCollectionPermissionLevel($content);
+                break;
+        }
+
+        return $content;
+    }
+
+    // New parameters
+
+    protected function addId(array $content, string $key): array
+    {
+        $content[$key] = 'unique()';
+        return $content;
+    }
+
+    protected function addCollectionPermissionLevel(array $content): array
+    {
+        $content['permission'] = 'document';
+        return $content;
+    }
+
+    // Deprecated parameters
+
+    protected function removeRules(array $content): array
+    {
+        unset($content['rules']);
+        return $content;
+    }
+
+    protected function removeOrderType(array $content): array
+    {
+        unset($content['orderType']);
+        return $content;
+    }
+
+    protected function removeOrderCast(array $content): array
+    {
+        unset($content['orderCast']);
+        return $content;
+    }
+
+    protected function removeParentProperties(array $content): array
+    {
+        unset($content['parentDocument']);
+        unset($content['parentProperty']);
+        unset($content['parentPropertyType']);
+        return $content;
+    }
+
+    // Modified parameters
+
+    protected function convertStatus(array $content): array
+    {
+        $content['status'] = $content['status'] === 2 ? false : true;
+        return $content;
+    }
+
+    protected function convertOrder(array $content): array
+    {
+        $content['orderAttributes'] = [ $content['orderField'] ];
+        $content['orderTypes'] = [ $content['orderType'] ];
+
+        unset($content['orderField']);
+        unset($content['orderType']);
+
+        return $content;
+    }
+
+    protected function convertQueries(array $content): array
+    {
+        $queries = [];
+
+        if(!empty($content['filters'])) {
+            foreach ($content['filters'] as $filter) {
+                $operators = ['=' => 'equal', '!=' => 'notEqual', '>' => 'greater', '<' => 'lesser', '<=' => 'lesserEqual', '>=' => 'greaterEqual'];
+                foreach ($operators as $operator => $operatorVerbose) {
+                    if (\str_contains($filter, $operator)) {
+                        $usedOperator = $operator;
+                        break;
+                    }
+                }
+
+                if(isset($usedOperator)) {
+                    [ $attributeKey, $filterValue ] = \explode($usedOperator, $filter);
+                    // TODO: String or not? Any way to figure out?
+                    $query = $attributeKey . '.' . $operators[$usedOperator] . '("' . $filterValue . '")';
+                    \array_push($queries, $query);
+                }
+            }
+        }
+
+        // TODO: Can we even migrate search? Which key? Is $id key OK?
```

Comment:
Not sure I understand.

Reply:
There used to be `search` parameter on `listDocuments` I am not sure what it was searching by and how we could achieve the same logic.. We cant, right? The only attribute indexed by default is `$id`, so that is the only place we could even attempt to do a serch.